src/ApplicationLogs/Store/States/ContractLogState.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2015-2023 The Neo Project.
+//
+// The Neo.Plugins.ApplicationLogs is free software distributed under the MIT software license,
+// see the accompanying file LICENSE in the main directory of the
+// project or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo;
+using Neo.IO;
+using Neo.Ledger;
+using Neo.SmartContract;
+
+namespace ApplicationLogs.Store.States
+{
+    public class ContractLogState : NotifyLogState, IEquatable<ContractLogState>
+    {
+        public UInt256 TransactionHash { get; private set; } = new();
+        public TriggerType Trigger { get; private set; } = TriggerType.All;
+
+        public static ContractLogState Create(Blockchain.ApplicationExecuted applicationExecuted, NotifyEventArgs notifyEventArgs, Guid[] stackItemIds) =>
+            new()
+            {
+                TransactionHash = applicationExecuted.Transaction?.Hash ?? new(),
+                ScriptHash = notifyEventArgs.ScriptHash,
+                Trigger = applicationExecuted.Trigger,
+                EventName = notifyEventArgs.EventName,
+                StackItemIds = stackItemIds,
+            };
+
+        #region ISerializable
+
+        public override int Size =>
+            TransactionHash.Size +
+            sizeof(byte) +
+            base.Size;
+
+        public override void Deserialize(ref MemoryReader reader)
```

Comment:
`Deserialize` content **always** come from a `Serialize` output?

Reply:
I'm overriding the `base` class of `NotifyLogState` in turn inherits `ISerialize`

Reply:
We initially create it 1st than serialize it, Hints the `Create` `static` method, above. But to answer question yes?

Reply:
@shargon any news?

Reply:
I need more time sorry